### PR TITLE
Set up Docker Compose and make command for running external services

### PR DESCRIPTION
**Pull request by guilload**
### Description
Set up  Docker Compose to manage our growing list of external services.

### How was this PR tested?
Ran `make docker-compose-up` and `QUICKWIT_ENV=local cargo test --all-features` locally.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -17,23 +17,24 @@ In that case, simply mark the PR with the tag [WIP] (standing for work in progre
...
 Contributors are required to sign a Contributor License Agreement.
 The process is simple and fast. Upon your first pull request, you will be prompted to
 [sign our CLA by visiting this link](https://cla-assistant.io/quickwit-inc/quickwit).
 
 # Development
 ## Setup & run tests
-1. Install docker https://docs.docker.com/engine/install/
-2. Install localstack https://github.com/localstack/localstack#installing
-3. Install awslocal https://github.com/localstack/awscli-local
-4. Prepare s3 bucket used in tests, execute `./quickwit-cli/tests/prepare_tests.sh`
```
Comment (fmassot) [inline CONTRIBUTING.md:31]: `prepare_tests.sh` seems to not be used anymore, you can remove it.

File: Makefile [hunk 2]
```diff
@@ -1,6 +1,18 @@
+DOCKER_SERVICES ?= all
+
 help:
 	@grep '^[^#[:space:]].*:' Makefile
 
+# Usage:
+# `make-docker-compose-up` starts all the services.
```
Comment (fmassot) [inline Makefile:7]: `make-docker-compose-up` or `make docker-compose-up` ?

not sure if it's a typo or a convention in a makefile.

File: CONTRIBUTING.md (hunk 1 above)
Comment (guilload) [reply CONTRIBUTING.md:31]: It is still used in CI. I will modify the CLI test suite in a follow-up PR, and then I'll get rid of it.

File: Makefile (hunk 2 above)
Comment (guilload) [reply Makefile:7]: A typo, thanks for pointing that out.

**Review by fmassot — APPROVED** [review]
LGTM.

**Pull request merged**